python/dgl/nn/pytorch/network_emb.py

```diff
@@ -0,0 +1,220 @@
+        print("real embeddings：" + str(len(self.word_frequency)))
+
+        self.u_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+        self.v_embeddings = nn.Embedding(self.word_count, self.emb_dimension, sparse=self.sparse)
+
+    def _initTableNegatives(self):
+        # get a table for negative sampling, if word with index 2 appears twice, then 2
+        # will be listed in the table twice.
+        pow_frequency = np.array(list(self.word_frequency.values())) ** 0.75
+        words_pow = sum(pow_frequency)
+        ratio = pow_frequency / words_pow
+        count = np.round(ratio * self.NEGATIVE_TABLE_SIZE)
+        word_list = list(self.word_frequency.keys())
+        for wid, c in enumerate(count):
+            self.negatives += [word_list[wid]] * int(c)
+        self.negatives = np.array(self.negatives)
+        np.random.shuffle(self.negatives)
+        self.sampling_prob = ratio
+
+    def initParameters(self):
+        """
+        Initialize the embedding parameters
+        """
+        initrange = 1.0 / self.emb_dimension
+        init.uniform_(self.u_embeddings.weight.data, -initrange, initrange)
+        init.constant_(self.v_embeddings.weight.data, 0)
+
+    def _getNegatives(self, size):
+        response = self.negatives[self.negpos:self.negpos + size]
+        self.negpos = (self.negpos + size) % len(self.negatives)
+        if len(response) != size:
+            return np.concatenate((response, self.negatives[0:self.negpos]))
+        return response
+
+    def _generate_sample(self, batches):
+        pair_catch = []
+        for batch in batches:
+            if len(batch) < 0:
+                continue
+            word_ids = [w for w in batch if w in self.word_frequency
+                        and np.random.rand() < self.discards[w]]
+
+            for i, u in enumerate(word_ids):
+                for j, v in enumerate(
+                        word_ids[max(i - self.window_size, 0):i + self.window_size]):
+                    assert u < self.word_count
+                    assert v < self.word_count
+                    if i == j:
+                        continue
+                    pair_catch.append((u, v, self._getNegatives(self.negative_samples_size)))
+
+        all_u = [u for u, _, _ in pair_catch]
+        all_v = [v for _, v, _ in pair_catch]
+        all_neg_v = [neg_v for _, _, neg_v in pair_catch]
+
+        return torch.LongTensor(all_u), torch.LongTensor(all_v), torch.LongTensor(all_neg_v)
+
+    def loader(self, **kwargs):
+        r"""Returns the data loader that contains center node ，positive context node，
+            and negative samples on the heterogeneous graph random walk.
+        Args:
+            **kwargs (optional): Arguments of
+                :class:`torch.utils.data.DataLoader`, such as
+                :obj:`batch_size`, :obj:`shuffle`, :obj:`drop_last` or
+                :obj:`num_workers`.
```

Comment:
You can reuse the doc of `kwargs` here: https://docs.dgl.ai/generated/dgl.dataloading.DataLoader.html#dgl.dataloading.DataLoader